kolibri/plugins/coach/assets/src/vue/groups-page/index.vue

```diff
@@ -1,71 +1,159 @@
+    <create-group-modal v-if="showCreateGroupModal"
       :className="className"
-      :classId="classId"/>
+      :classId="classId" />
 
-    <div v-for="group in groups">
-      <h2>{{ group.name }}</h2>
-      <!--{{ $tr('numLearners', {count: group.users.length}) }} -->
-    </div>
+    <rename-group-modal v-if="showRenameGroupModal"
+      :className="className"
+      :classId="classId"
+      :groupName="selectedGroup.name"
+      :groupId="selectedGroup.id" />
+
+    <delete-group-modal v-if="showDeleteGroupModal"
+      :className="className"
+      :classId="classId"
+      :groupName="selectedGroup.name"
+      :groupId="selectedGroup.id" />
+
+    <move-learners-modal v-if="showMoveLearnersModal"
+      :className="className"
+      :classId="classId"
+      :groupName="selectedGroup.name"
+      :groupId="selectedGroup.id"
+      :groups="groups"
+      :usersToMove="usersToMove"
+      :isUngrouped="isUngrouped" />
+
+    <group-section v-for="group in groups"
+      :group="group"
+      :className="className"
+      :classId="classId"
+      @rename="openRenameGroupModal"
+      @delete="openDeleteGroupModal"
+      @move="openMoveLearnersModal" />
+
+    <group-section :group="ungroupedUsersObject"
+      :className="className"
+      :classId="classId"
+      :isUngrouped="true"
+      @move="openMoveLearnersModal" />
   </div>
 
 </template>
 
 
 <script>
 
-  const actions = require('../../actions');
+  const groupActions = require('../../group-actions');
   const constants = require('../../state/constants');
-
+  const differenceWith = require('lodash.differencewith');
 
   module.exports = {
     $trNameSpace: 'coachGroupsPage',
     $trs: {
       groups: 'Groups',
       newGroup: 'New group',
-      numLearners: '{count, number, integer} {count, plural, one {Learner} other {Learners}}'
+      ungrouped: 'Ungrouped',
     },
     components: {
       'icon-button': require('kolibri.coreVue.components.iconButton'),
       'create-group-modal': require('./create-group-modal'),
+      'group-section': require('./group-section'),
+      'rename-group-modal': require('./rename-group-modal'),
+      'delete-group-modal': require('./delete-group-modal'),
+      'move-learners-modal': require('./move-learners-modal'),
+    },
+    data() {
+      return {
+        selectedGroup: { name: '', id: '' },
+        usersToMove: [],
+        isUngrouped: false,
+      };
     },
     computed: {
       showCreateGroupModal() {
         return this.modalShown === constants.Modals.CREATE_GROUP;
       },
+      showRenameGroupModal() {
+        return this.modalShown === constants.Modals.RENAME_GROUP;
+      },
+      showDeleteGroupModal() {
+        return this.modalShown === constants.Modals.DELETE_GROUP;
+      },
+      showMoveLearnersModal() {
+        return this.modalShown === constants.Modals.MOVE_LEARNERS;
+      },
+      groupedUsers() {
+        const groupedUsers = [];
+        this.groups.forEach(group => {
```

Comment:
Do the users not have a group ID associated with them? If so, couldn't we just do a `forEach` on all users and get the ones that have no group ID?